src/averageAges.js

```diff
@@ -1,64 +1,52 @@
 'use strict';
 
-/**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
- * @param {object[]} people
- * @param {number} century - optional
- *
- * @return {number}
- */
+const SEX_MALE = 'm';
+const SEX_FEMALE = 'f';
+
+function calculateTotalAge(people) {
+  return people.reduce((sum, person) => sum + (person.died - person.born), 0);
+}
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const filterMen = people.filter(({ sex, died }) =>
+    sex === SEX_MALE && (century
+      ? Math.ceil(died / 100) === century
+      : true
+    ));
+
+  const totalAgeMen = calculateTotalAge(filterMen);
+
+  return totalAgeMen / filterMen.length;
```

Comment:
You make similar calculations for `women`, so you should transfer this division to the `calculateTotalAge` function and rename it to `calculateAverageAge`.